### DISCUSSION TITLE
How can I write a method in reactiveui that works whenever I 'get' a property on the view?

### DISCUSSION THREAD

**Discussion by highspeeder** [AvaloniaUI/Avalonia#9222] [Q&A]
How can I write a method in reactiveui that works whenever I 'get' a property on the view?
I could write it directly in the get of the property, but I was wondering if there is a reactiveui-only way.

**Comment by maxkatz6** [comment]
No, there is no "reactiveui-only" way as far as I know. You might ask it in ReactiveUI repository.

Furthermore, it's not considered as a good practice to have getter methods with side effects.

**Reply by highspeeder** [reply]
So how do I implement a method that works whenever I do a getter?

**Reply by maxkatz6** [reply]
What do you want to do?
And why do you want to rely on view calling a getter?

Again, you can modify your getter, but it doesn't sound like a good idea in general.

**Reply by highspeeder** [reply]
I want to update the collection when I "get" it from the view.

**Reply by maxkatz6** [reply]
If you need a lazy collection, you can do something like this:
```csharp
private ObservableCollection<object>? _items;
public ObservableCollection<object> Items => _items ??= CreateInitialCollection();
```
So, this way `_items` field will be initialized once on first read. It is a pretty common .NET pattern to delay initialization of properties. Alternatively, you can use `Lazy<T>` class.
But keep in mind that View will cache this collection. If you need to update this collection on some **specific** event, like opening the page or pressing the button, then you should handle these events and update collection explicitly.

**Comment by timunie** [comment]
You may want to listen to changes of the property. ReactiveUI has WhenAnyValue for this.
